lib/rs-address.ts

```diff
@@ -0,0 +1,375 @@
+/*
+    NXT address class, extended version (with error guessing).
+
+    Version: 1.0, license: Public Domain, coder: NxtChg (admin@nxtchg.com).
+*/
+
+export class RSAddress {
+    codeword = [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+    syndrome = [0, 0, 0, 0, 0];
+
+    gexp = [1, 2, 4, 8, 16, 5, 10, 20, 13, 26, 17, 7, 14, 28, 29, 31, 27, 19, 3, 6, 12, 24, 21, 15, 30, 25, 23, 11, 22, 9, 18, 1];
+    glog = [0, 0, 1, 18, 2, 5, 19, 11, 3, 29, 6, 27, 20, 8, 12, 23, 4, 10, 30, 17, 7, 22, 28, 26, 21, 25, 9, 16, 13, 14, 24, 15];
+
+    cwmap = [3, 2, 1, 0, 7, 6, 5, 4, 13, 14, 15, 16, 12, 8, 9, 10, 11];
+
+    alphabet = '23456789ABCDEFGHJKLMNPQRSTUVWXYZ';
+    //var alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ345679';
+
+    guess: string[] = [];
+
+    ginv(a: any) {

```

Comment:
a is a number, no?

Reply:
rs-address is one of the 2 files (alongside curve25519 ) that are 3rd party and had no type definition. I just changed it to ts enough to make it work. I tried to add more types but I realized that I don;t know most types and was to much work for an external, already tested library.

For now I will not adapt it, but I took a note for when I have more free time.
This should be safe enough as this library is not exposed to be used directly from outside the SDK, but only an internal helper to quickly translate public addresses to RS format (GMD-xxxx-xxxx-xxxx-xxxx). Before integrating this library we would had needed to make a call to the node to get the RS format from public key.